### PR TITLE
Feature/dynamic module load

### PR DESCRIPTION
**Pull request by bigbitbus**
# Description
Load Opta modules dynamically instead of loading all of them in the layer.py file.

# Safety checklist
* [X] This change is backwards compatible and safe to apply by existing users
* [X] This change will NOT lead to data loss
* [X] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
Manual testing

### CODE REVIEW

File: opta/registry.py [hunk 1]
```diff
@@ -50,7 +51,7 @@ def make_registry_dict() -> Dict[Any, Any]:
             alt_cloudname = cloud
         cloud_dict["modules"] = {**_make_module_registry_dict(module_path, alt_cloudname)}
         registry_dict[cloud] = cloud_dict
-
+    open("/tmp/registry.json", "w").write(json.dumps(registry_dict))  # nosec
```
Comment (rabbitfang) [inline opta/registry.py:54]: Was this meant to be left in?

File: opta/layer.py [hunk 2]
```diff
@@ -75,6 +35,55 @@
 from opta.utils import deep_merge, hydrate, logger, yaml
 from opta.utils.dependencies import validate_installed_path_executables
 
+PROCESSOR_DICT: Dict[str, str] = {
```
Comment (rabbitfang) [inline opta/layer.py:38]: Can you please add a unit test that checks that all of the processors in this dict exist? Something like:
```python
for processor in PROCESSOR_DICT.values():
    pymodule_path = generate_pymodule_path(processor)
    importlib.import_module(pymodule_path)
```

**Review by rabbitfang — CHANGES_REQUESTED** [review]

File: opta/registry.py (hunk 1 above)
Comment (bigbitbus) [reply opta/registry.py:54]: no, it was debug info, removed.

File: opta/layer.py (hunk 2 above)
Comment (bigbitbus) [reply opta/layer.py:38]: Added a test to check if each module in the dictionary is actually in the directories.

**Comment by bigbitbus** [comment]
@rabbitfang addressed the comments in the review. @juandiegopalomino as discussed when we use this in the new module api we will try to automatically infer the mapping instead of the dictionary. For now with @rabbitfang suggestion I added a test to fail if the module code does not exist at the expected location in the module directory.

File: tests/test_layer.py [hunk 3]
```diff
@@ -322,3 +376,11 @@ def test_child(self, mocker: MockFixture):
...
             ]
         )
+
+    def test_check_module_dirs_exist(self):
+        from opta.layer import PROCESSOR_DICT, get_processor_class
+
+        for opta_module, processor in PROCESSOR_DICT.items():
+            module_class = get_processor_class(opta_module)
+            assert module_class.__name__ == processor
+        print("done")
```
Comment (rabbitfang) [inline tests/test_layer.py:386]:
```suggestion
```

**Review by rabbitfang — APPROVED** [review]
Everything else LGTM

**Comment by bigbitbus** [comment]
@rabbitfang added your commit.

**Pull request merged**